src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -25,18 +22,24 @@ internal sealed partial class EditorConfigDocumentOptionsProvider : IDocumentOpt
         /// </summary>
         private readonly Dictionary<DocumentId, Task<ICodingConventionContext>> _openDocumentContexts = new Dictionary<DocumentId, Task<ICodingConventionContext>>();
 
+        private readonly Workspace _workspace;
+        private readonly IAsynchronousOperationListener _listener;
         private readonly ICodingConventionsManager _codingConventionsManager;
         private readonly IErrorLoggerService _errorLogger;
 
-        internal EditorConfigDocumentOptionsProvider(Workspace workspace)
+        internal EditorConfigDocumentOptionsProvider(Workspace workspace, ICodingConventionsManager codingConventionsManager, IAsynchronousOperationListenerProvider listenerProvider)
         {
-            _codingConventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
+            _workspace = workspace;
+            _listener = listenerProvider.GetListener(FeatureAttribute.SolutionCrawler);
```

Comment:
add new FeatureAttribute

Reply:
It seemed like `Workspace` or `SolutionCrawler` would most accurately reflect the impact of this change on existing code that waits for asynchronous operations to complete. It's not clear what the difference between the two is, but it seems unlikely that we'd want something else because a bunch of call sites would suddenly need to be updated.

Reply:
what I meant was, you just add FeatureAttributes.EditorConfigOptionProvider and use that. rather than use some random features name here which has nothing to do with either Workspace or SolutionCrawler.

Reply:
It looks like `FeatureAttribute.Workspace` is the one we want for this to work without changes:

https://github.com/dotnet/roslyn/blob/d0ffbbd6ddfec1cd588b9c4e0f08e369267910c2/src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs#L191

Reply:
workspace is supposed only for ones raised by workspace. this is raised by editorconfigoption provider.
I think you should just add editorconfigoptionprovider to featureattribute and add them to ones needed. if you think formatdocument needs to wait editorconfigoptionprovider, explicitly add there.